src/main/java/seedu/address/logic/commands/module/ModuleDeleteCommand.java

```diff
@@ -22,31 +23,56 @@
     public static final String MESSAGE_USAGE = COMMAND_GROUP_MOD + " " + COMMAND_WORD_DELETE
             + ": Deletes the module identified by the index number used in the displayed module list.\n"
             + "Parameters: INDEX (must be a positive integer)\n"
-            + "Example: " + COMMAND_GROUP_MOD + " " + COMMAND_WORD_DELETE + " 1";
+            + "Example: " + COMMAND_GROUP_MOD + " " + COMMAND_WORD_DELETE + " 1\n"
+            + "Parameters: MODULE_CODE (must be a valid module code)\n"
+            + "Example: " + COMMAND_GROUP_MOD + " " + COMMAND_WORD_DELETE + " CS4215";
 
     public static final String MESSAGE_DELETE_MODULE_SUCCESS = "Deleted module: %1$s";
 
     private final Index targetIndex;
+    private final ModuleCode moduleCode;
 
     /**
      * Creates a ModuleDeleteCommand to delete the module the specified {@code index}.
      */
     public ModuleDeleteCommand(Index targetIndex) {
+        requireNonNull(targetIndex);
+
         this.targetIndex = targetIndex;
+        this.moduleCode = null;
+    }
+
+    public ModuleDeleteCommand(ModuleCode moduleCode) {
+        requireNonNull(moduleCode);
+
+        this.moduleCode = moduleCode;
+        this.targetIndex = null;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<Module> lastShownList = model.getFilteredModuleList();
+        Module moduleToDelete = null;
 
-        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+        if (targetIndex != null && targetIndex.getZeroBased() >= lastShownList.size()) {
             throw new CommandException(Messages.MESSAGE_INVALID_MODULE_DISPLAYED_INDEX);
+        } else if (targetIndex != null) {
+            moduleToDelete = lastShownList.get(targetIndex.getZeroBased());
+        } else { // index == null
+            assert moduleCode != null;
+            for (Module mod : lastShownList) {
+                if (mod.getModuleCode().equals(moduleCode)) {
+                    moduleToDelete = mod;
+                    break;
+                }
+            }
         }
 
-        Module moduleToDelete = lastShownList.get(targetIndex.getZeroBased());
+        assert moduleToDelete != null;
```

Comment:
Maybe can throw exception instead like this module does not exist

Reply:
yup yup